Code/Mantid/MantidQt/CustomInterfaces/inc/MantidQtCustomInterfaces/ReflGenerateNotebook.h

```diff
@@ -0,0 +1,125 @@
+#ifndef MANTID_CUSTOMINTERFACES_REFLGENERATENOTEBOOK_H
+#define MANTID_CUSTOMINTERFACES_REFLGENERATENOTEBOOK_H
+
+/** @class ReflGenerateNotebook
+
+    This class creates ipython notebooks from the ISIS Reflectometry
+    (Polref) interface
+
+    Copyright &copy; 2007-8 ISIS Rutherford Appleton Laboratory, NScD Oak Ridge
+    National Laboratory & European Spallation Source
+
+    This file is part of Mantid.
+
+    Mantid is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    Mantid is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+    File change history is stored at: <https://github.com/mantidproject/mantid>.
+    Code Documentation is available at: <http://doxygen.mantidproject.org>
+    */
+
+#include "MantidQtCustomInterfaces/QReflTableModel.h"
+#include "MantidKernel/System.h"
+
+#include <set>
+#include <map>
+#include <string>
+#include <sstream>
+
+
+namespace MantidQt {
+  namespace CustomInterfaces {
+
+
+    // Column numbers to find data in model
+    struct ColNumbers {
+
+      ColNumbers(const int runs_column, const int transmission_column, const int options_column,
+                 const int angle_column, const int qmin_column, const int qmax_column,
+                 const int dqq_column, const int scale_column, const int group_column)
+        : runs(runs_column), transmission(transmission_column), options(options_column), angle(angle_column),
+          qmin(qmin_column), qmax(qmax_column), dqq(dqq_column), scale(scale_column), group(group_column) {}
+
+      const int runs;
+      const int transmission;
+      const int options;
+      const int angle;
+      const int qmin;
+      const int qmax;
+      const int dqq;
+      const int scale;
+      const int group;
+    };
+
+    std::string plot1DString(const std::vector<std::string> & ws_names,
+                             const std::string & title);
+
+    std::string tableString(QReflTableModel_sptr model, ColNumbers col_nums, const std::set<int> & rows);
+
+    std::string titleString(const std::string & wsName);
+
+    std::tuple<std::string, std::string>
```

Comment:
Didn't know we can use tuple. This might be an issue with older compilers, but Jenkins will tell us so.